Documentation/design-docs/DotNetCore-SharedPackageStore.md

```diff
@@ -0,0 +1,218 @@
+# .NET Core - Shared Package Store
+
+## Introduction
+To enable sharing of assemblies among all machine-wide .NET Core applications, a centralized shared assembly store is needed. It enables applications to be trimmed of the shared assemblies and frees the apps from carrying them. This document will describe such an assembly store and its lookup that applications can take advantage of during development and their deployment. It also describes commands that will augment `dotnet` to compose shared package store entries and publishing mechanisms through which apps can filter assemblies to reduce disk usage.
+
+### Packages Store
+
+The package store can be either a global system-wide folder or a user local folder:
+
++ **Global**:
+    - The `dotnet` root location -- on Windows, the folder is located in `C:\Program Files (x86)\`. See layout below.
+    - In the `packages/install` folder we expect packages to be installed ONLY through platform installers like MSI, pkg, deb, apt-get etc.
+    - The package layout composed with the `dotnet cache` command (details follow) are expected to be unzipped directly into the `packages` folder. Note, that this unzip step is a manual action.
+
+```
+    - dotnet.exe
+    - shared
+        - netcoreapp2.0
+            + 2.0.0-preview2-00001
+    - packages
+        - install
+            + refs
+            + netcoreapp2.0
+            + netcoreapp2.1
+        + refs
+        + netcoreapp2.0
+        + netcoreapp2.1  
+```
+
+The layout within `netcoreapp*` folders is a NuGet cache layout.
+
++ **User**: The user local location can be used to override the global location when probing assemblies and uses a very similar layout, except that the `packages/install` folder is NOT supported in the user profile. It is only supported in the global location.
+
+```
+    - %USERPROFILE% or $HOME/
+        - .dotnet
```

Comment:
The '.' has become the convention for such tool-related folders, from ".git" to local IDE settings, so it seems appropriate here too. Such folders do indeed become hidden on Linux, but that's generally known by Linux users who know how to make them appear when needed.
User profile should be the default, but there can be an option for system-wide, that would require admin rights.
Does this help?